src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+  tooltip: {
+    background: 'rgba(0, 0, 0, .8)',
+    color: 'white',
+    borderRadius: '4px',
+    pointerEvents: 'none',
+  },
+  tooltipKey: {
+    display: 'inline-block',
+    width: '10px',
+    height: '10px',
+    marginRight: '10px',
+  },
+};
+
+function Footer({ dataset, index, currData }) {
+  const prevData = dataset[index - 1].value;
+  const prevRevision = dataset[index - 1].revision;
+  const delta = (currData - prevData).toFixed(2);
+  const deltaPercentage = ((currData / prevData - 1) * 100).toFixed(2);
+  const currRevision = dataset[index].revision;
+  const hgURL = `https://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=${prevRevision}&tochange=${currRevision}`;
+
+  return (
+    <React.Fragment>
+      <div>
+        Δ {delta} ({deltaPercentage} %)
+      </div>
+      <div>
+        <a href={hgURL}>{currRevision.slice(0, 12)}</a>
+      </div>
+    </React.Fragment>
+  );
+}
+
+function CustomTooltip({ classes, tooltipModel, series }) {
+  if (tooltipModel.opacity === 0) return null;
+
+  const { xLabel, index, datasetIndex } = tooltipModel.dataPoints[0];
+  const dataset = series[datasetIndex].data;
+  const seriesLabel = series[datasetIndex].label;
+  const labelColors = tooltipModel.labelColors[0];
+  const currData = tooltipModel.dataPoints[0].yLabel;
+  const higherIsBetter = !series[datasetIndex].meta.lower_is_better;
+  const higherOrLower = higherIsBetter ? 'higher is better' : 'lower is better';
+  const paddingStyle = {
```

Comment:
Find out what this `paddingStyle` is and copy it into `styles` at the top of the file. Remove `paddingStyle`

Reply:
`paddingStyle`are options that are fetched from `tooltip.xPadding` and `tooltip.yPadding` chart options. By default it's value is 6px, if we set a fixed padding value to our custom tooltip (i.e., setting 6px in the `const styles = {...}`), then if in the future someone wants to set a padding inside `perfherderFormatter`, like this :
```jsx
// perherderFormatter.js
...
const generateInitialOptions = series => {
  ...
  return {
    ... 
    tooltips: {
      xPadding: '8px';
      yPadding: '4px';
      ...
      custom(tooltipModel) {...}
    }
  }
}
``` 
it **won't work** and will ignore that padding. Is there a way that we can set dynamically those values inside `const styles = {...} `? Because I can't see it. Or if you want to set fixed values as I mentioned before, we will lose a functionality in the chart options.

Reply:
Yes, I am aware these types of styling "features" will be lost; that is the side effect of having a CustomTooltip.  

There should be one place that has all the styling information, and it should be the `const styles` at the top of the file.  Setting stylings in another place, like the `tooltips` substructure, runs the risk of being confusing.  Suppose someone does set `xPadding: '4px'`, and I try to set the `padding` in the `const styles`: I am in for some pain as I try to figure out why my styling is not working.